packages/iTwinUI-react/src/core/Table/actionHandlers/selectHandler.ts

```diff
@@ -56,6 +56,20 @@ export const onSelectHandler = <T extends Record<string, unknown>>(
   onSelect?.(selectedData, newState);
 };
 
+export const onToggleHandler = <T extends Record<string, unknown>>(
```

Comment:
I don't know a lot about this - why did you make a new function instead of updating `onSelectHandler` to include the `action` property? I see that other functions are using it, but maybe they could be updated too, or it could be optional?

Reply:
We want to update `lastSelectedRowId` only on checkbox click and row click, but not on shift click.

Checkbox click calls onSelectHandler() ([L495](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-9edd44b431060c27e0260d286ee188bdf901225b1ee84ff6540f9d796dd0f999L495)). So we could update lastSelectedRowId in onSelectHandler(). But the problem is that onShiftSelectHandler() also calls onSelectHandler() ([L101](https://github.com/iTwin/iTwinUI-react/pull/889/files#diff-772f25aaf3cd0aa17d5900a4252050c32c48b2efe18b4f6db0d84df08f96cc0bR101)). So we can't update checkbox click in onSelectHandler() since then lastSelectedRowId also updates on shift click.

Thus, checkbox click now calls `onToggleHander()` and updates lastSelectedRowId there. Shift select does not update lastSelectedRowId since onShiftSelectHandler() still calls `onSelectHandler()` and not `onToggleHandler()`